### PR TITLE
[feature](function) support ip function is_ipv4_compat, is_ipv4_mapped

### PR DESCRIPTION
**Pull request by superdiaodiao**
## Proposed changes

Issue Number: https://github.com/apache/doris/issues/19624

**MySQL**:
image

image

**Doris**:
image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run pipelinex_p0

**Comment by superdiaodiao** [comment]
run pipelinex_p0

**Comment by superdiaodiao** [comment]
run buildall

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -969,4 +930,114 @@ class FunctionIPv6CIDRToRange : public IFunction {
...
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return make_nullable(std::make_shared<DataTypeUInt8>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return true; }
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:946]: no need override the function， the default value is `true`

File: be/src/vec/functions/function_ip.h [hunk 2]
```diff
@@ -969,4 +930,114 @@ class FunctionIPv6CIDRToRange : public IFunction {
...
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:1001]: no need override the function， the default value is `true`

File: be/src/vec/functions/function_ip.h [hunk 3]
```diff
@@ -969,4 +930,114 @@ class FunctionIPv6CIDRToRange : public IFunction {
...
+                            column->get_name(), get_name());
+        size_t col_size = col_in->size();
+        auto col_res = ColumnUInt8::create(col_size, 0);
+        auto null_map = ColumnUInt8::create(col_size, 0);
+
+        for (size_t i = 0; i < col_size; ++i) {
+            auto& col_res_data = col_res->get_data();
+            auto& null_map_data = null_map->get_data();
+
+            if (col_in->is_null_at(i)) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:1020]: why this function is `always nullable`, i think better be `default`

File: be/src/vec/functions/function_ip.h (hunk 1 above)
Comment (superdiaodiao) [reply be/src/vec/functions/function_ip.h:946]: okay, i fogot to delete, i will delete it

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Ipv6StringToNum.java [hunk 4]
```diff
@@ -35,7 +35,7 @@
  * scalar function ipv6_string_to_num
  */
 public class Ipv6StringToNum extends ScalarFunction
-        implements BinaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Ipv6StringToNum.java:38]: why change the code? seems not need, and not the impl function

**Comment by superdiaodiao** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by yiguolei — DISMISSED** [review]

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wm1581066 — APPROVED** [review]
LGTM

**Pull request merged**